src/include/sof/trace.h

```diff
@@ -205,70 +244,75 @@ typedef void(*log_func)();
  * associated with it, to save on transmission. It is then placed by the linker
  * in .static_log_entries section of ELF.
  */
-#define _DECLARE_LOG_ENTRY(lvl, format, comp_id, params)\
-	__attribute__((section(".static_log." #lvl)))	\
-	static const struct {				\
-		uint32_t level;				\
-		uint32_t component_id;			\
-		uint32_t params_num;			\
-		uint32_t line_idx;			\
-		uint32_t file_name_len;			\
-		const char file_name[sizeof(__FILE__)];	\
-		uint32_t text_len;			\
-		const char text[sizeof(format)];	\
-	} log_entry = {					\
-		lvl,					\
-		comp_id,				\
-		params,					\
-		__LINE__,				\
-		sizeof(__FILE__),			\
-		__FILE__,				\
-		sizeof(format),				\
-		format					\
+#define _DECLARE_LOG_ENTRY(lvl, format, comp_class, params, ids)\
+	__attribute__((section(".static_log." #lvl)))		\
+	static const struct {					\
+		uint32_t level;					\
+		uint32_t component_class;			\
+		uint32_t has_ids;				\
```

Comment:
Where do we encode id for comp and pipeline here. I was expecting to see pipeline_id and component_id variables ?

Reply:
Component id and Pipeline id are runtime variables, so they cannot be part of log_entry. They are passed through log_entry_header.